src/synchronizer.ts

```diff
@@ -23,34 +26,74 @@ export class Synchronizer {
     } else {
       const failMessage = `Event '${context.eventName}' is not expected. Failing.`;
       this.logger.warning(failMessage);
       throw new Error(failMessage);
     }
   }
 
-  async updateAllIssues(excludeClosed: boolean = false): Promise<boolean> {
+  /**
+   * Gets the field node data ids to set custom fields
+   * This method will fail if the field or value are not available.
+   * @param project Project node data. Should be obtained from project kit
+   * @param customField key value pair with the names of the fields. Not case sensitive
+   * @returns Returns a key value pair with the node id of both the field and the value or null if the application threw an error
+   */
+  private async getCustomFieldNodeData(project: NodeData, customField?: FieldValues): Promise<FieldValues | null> {
+    if (!customField) {
+      return null;
+    }
+
+    try {
+      return await this.projectKit.fetchProjectFieldNodeValues(project, customField);
+    } catch (e) {
+      this.logger.error("Failed fetching project values.");
+      throw e;
+    }
+  }
+
+  private async updateAllIssues(excludeClosed: boolean = false, customField?: FieldValues): Promise<boolean> {
     const issues = await this.issueKit.getAllIssues(excludeClosed);
     if (issues?.length === 0) {
       this.logger.notice("No issues found");
       return false;
     }
     this.logger.info(`Updating ${issues.length} issues`);
-    const issueAssigment = await this.projectKit.assignIssues(issues);
-    return issueAssigment.every((s) => s);
+
+    const projectNode = await this.projectKit.fetchProjectData();
+    const customFieldNodeData = await this.getCustomFieldNodeData(projectNode, customField);
+    const issuesAssigmentPromises = issues.map((issue) => this.projectKit.assignIssue(issue, projectNode));
+    const issuesCardIds = await Promise.all(issuesAssigmentPromises);
+    this.logger.debug(`Finished assigning ${issuesCardIds.length} issues`);
+    if (customFieldNodeData) {
+      this.logger.debug("Found custom field node data for " + JSON.stringify(customField));
+      const assignCustomFieldPromise = issuesCardIds.map((ici) =>
+        this.projectKit.changeIssueStateInProject(ici, projectNode, customFieldNodeData),
+      );
+      await Promise.all(assignCustomFieldPromise);
+    }
+
+    return true;
   }
 
-  async updateOneIssue(issue: Issue): Promise<boolean> {
-    return await this.projectKit.assignIssue(issue);
+  private async updateOneIssue(issue: Issue, customField?: FieldValues): Promise<boolean> {
+    const projectNode = await this.projectKit.fetchProjectData();
+    const customFieldNodeData = await this.getCustomFieldNodeData(projectNode, customField);
+    const issueCardId = await this.projectKit.assignIssue(issue, projectNode);
+    if (customFieldNodeData) {
+      await this.projectKit.changeIssueStateInProject(issueCardId, projectNode, customFieldNodeData);
+    }
+
+    return true;
```

Comment:
Sounds like it should return `void`. Especially that error conditions here are throwing.

Reply:
I [return false here](https://github.com/paritytech/github-issue-sync/pull/51/files#diff-c2d0c14d65663133af3eaaf1f4426cc8c95ee3e70b9bd807d40f11c6138dfe0cR69) when no issues could be found.

Is it ok if I keep it there for future reference? Or should I simply remove it and replace it with empty `return`?

Reply:
I'm against returning booleans for indication of success/error.  
This is both semantically confusing ("why does an `update` function return boolean?"), and very unspecific (how would a callee interpret `false`?)

How about using one of following: 
* return `void` with throwing an error in case there is one. The usual approach in JS/TS world.
* return a union type of something like `undefined | Error`, or `Ok | Err` from `opstooling-js`. More rust-like, but without exceptions.

Reply:
Will do then.

> I'm against returning booleans for indication of success/error.
I think I'm very old fashioned with having a `int main()` which returns 0 on success haha.

Reply:
> > I think I'm very old fashioned with having a `int main()` which returns 0 on success haha.

That ideology is actually built on error codes, where 0 stands for success, and other values often having special meaning, indicating the type of error.  
But yeah, who didn't do `process.exit(1)`, right? ¯\\\_(ツ)_/¯